zeppelin-zengine/src/main/java/org/apache/zeppelin/notebook/repo/S3NotebookRepo.java

```diff
@@ -94,33 +96,30 @@ public S3NotebookRepo(ZeppelinConfiguration conf) throws IOException {
 
     // always use the default provider chain
     AWSCredentialsProvider credentialsProvider = new DefaultAWSCredentialsProviderChain();
-    CryptoConfiguration cryptoConf = null;
+    CryptoConfiguration cryptoConf = new CryptoConfiguration();
     String keyRegion = conf.getS3KMSKeyRegion();
 
     if (StringUtils.isNotBlank(keyRegion)) {
-      cryptoConf = new CryptoConfiguration();
       cryptoConf.setAwsKmsRegion(Region.getRegion(Regions.fromName(keyRegion)));
     }
+
+    ClientConfiguration cliConf = createClientConfiguration();
     
     // see if we should be encrypting data in S3
     String kmsKeyID = conf.getS3KMSKeyID();
     if (kmsKeyID != null) {
       // use the AWS KMS to encrypt data
       KMSEncryptionMaterialsProvider emp = new KMSEncryptionMaterialsProvider(kmsKeyID);
-      if (cryptoConf != null) {
-        this.s3client = new AmazonS3EncryptionClient(credentialsProvider, emp, cryptoConf);
-      } else {
-        this.s3client = new AmazonS3EncryptionClient(credentialsProvider, emp);
-      }
+      this.s3client = new AmazonS3EncryptionClient(credentialsProvider, emp, cliConf, cryptoConf);
```

Comment:
The `if` branch is removed for the simplicity but the internal process of each branch is not change because `new CryptoConfiguration()` is called in the `new AmazonS3EncryptionClient(credentialsProvider, emp)` constructor eventually.